src/wgpu/wgpu-device.cpp

```
@@ -232,6 +238,14 @@ Result DeviceImpl::initialize(const DeviceDesc& desc, BackendImpl* backend)
         limits.maxFramebufferDimensions[1] = m_ctx.limits.maxTextureDimension2D;
         limits.maxFramebufferDimensions[2] = m_ctx.limits.maxTextureArrayLayers;
         limits.maxShaderVisibleSamplers = m_ctx.limits.maxSamplersPerShaderStage;
+
+        if (m_ctx.features.count(WGPUFeatureName_Subgroups) && subgroupMinSize > 0 &&
+            subgroupMinSize != WGPU_LIMIT_U32_UNDEFINED && subgroupMaxSize > 0 &&
+            subgroupMaxSize != WGPU_LIMIT_U32_UNDEFINED)
+        {
```

Cppcheck (2.20.0)

[style] The function 'emplace_back' is never used.

(unusedFunction)